src/solver/type_getters.jl

```diff
@@ -291,13 +291,20 @@ function get_spaces_restart(Y)
     return (; center_space, face_space)
 end
 
-function get_state_restart(config::AtmosConfig)
+function get_state_restart(config::AtmosConfig, restart_file, atmos_model_hash)
     (; parsed_args, comms_ctx) = config
-    restart_file = parsed_args["restart_file"]
+
     @assert !isnothing(restart_file)
     reader = InputOutput.HDF5Reader(restart_file, comms_ctx)
     Y = InputOutput.read_field(reader, "Y")
     t_start = InputOutput.HDF5.read_attribute(reader.file, "time")
+    atmos_model_hash_in_restart =
+        InputOutput.HDF5.read_attribute(reader.file, "atmos_model_hash")
+    if atmos_model_hash_in_restart != atmos_model_hash
+        error(
+            "Restart file $(restart_file) was constructed with a different AtmosModel",
+        )
```

Comment:
I'm wondering if we can do better exception handling here. This might be a fine crude start, but it might be nice to serialize the objects we're restarting, and then a check like this could print the differences so that users better understand why this might fail.

Reply:
Yes, I agree. My approach is to start with a restrict, but verified-to-be-working case, and move towards more general and flexible restarts.

This line would also probably disallow running with different versions of ClimaAtmos, which is not ideal.